### PR TITLE
Bump version to 4.5.5

### PR DESCRIPTION
**Pull request by traversaro**
I tried to use the existing `libopencv` 4.5.3 package with Visual Studio 2022, but it does not work as the `OpenCVConfig.cmake` file has some logic that needed to be update explicity for Visual Studio 2022 (see https://github.com/robotology/robotology-superbuild/pull/995#issuecomment-1010144950). As `4.5.5`  is the latest release, I updated directly to the last one.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ package:
 source:
   - url: https://codeload.github.com/opencv/opencv/tar.gz/{{ version }}
     fn: opencv-{{ version }}.tar.gz
-    sha256: 77f616ae4bea416674d8c373984b20c8bd55e7db887fd38c6df73463a0647bab
+    sha256: a1cfdcf6619387ca9e232687504da996aaa9f7b5689986b8331ec02cb61d28ad
     patches:
       - 20386-SetTotalBytesLimit-one-arg.patch
   - url: https://codeload.github.com/opencv/opencv_contrib/tar.gz/{{ version }}
```
Comment (hmaarrfk) [inline recipe/meta.yaml:36]: I think the contrib sha256 needs to be updated too.

Comment (traversaro) [reply recipe/meta.yaml:36]: Done in https://github.com/conda-forge/opencv-feedstock/pull/296/commits/6396c6d3417b8c4a82e7759c52d4f62943cdb174 .

**Comment by traversaro** [comment]
Thanks @hmaarrfk for removing the only existing patch, indeed this can be dropped thanks to https://github.com/opencv/opencv/pull/20386 .

**Comment by hmaarrfk** [comment]
Nice. yeah. tryinf to help with other config when I have time too.

**Comment by hmaarrfk** [comment]
Seems like we need to add a similar tesseract line for windows. are you able to do that?

**Comment by traversaro** [comment]
> Seems like we need to add a similar tesseract line for windows. are you able to do that?

Done: https://github.com/traversaro/opencv-feedstock/commit/04098806fd6f0e1ba4337504738bcf79b6588c7d .

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by traversaro** [comment]
fyi @wolfv @Tobias-Fischer as this could result in requiring rebuild on the RoboStack side.

**Comment by traversaro** [comment]
@hmaarrfk this will trigger a new CI, but probably we can remove the file https://github.com/traversaro/opencv-feedstock/blob/patch-2/recipe/20386-SetTotalBytesLimit-one-arg.patch, is that ok for you @hmaarrfk ?

**Comment by hmaarrfk** [comment]
I know the windows builds are slow, but thye are often the ones that fail for some random reason. Trying to ensure they pass before breaking things in the system.

**Comment by traversaro** [comment]
Thanks @hmaarrfk !

**Comment by traversaro** [comment]
@hmaarrfk not sure if you saw it, but in the post-merge CI job there was a spurious failure: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=439846&view=logs&jobId=4b3757f9-99c5-5891-6c0c-93d7030cdf7d .

**Comment by wolfv** [comment]
I tried pressing the rerun button

**Comment by hmaarrfk** [comment]
I did not see the failure. I think when yo urestarted it destroyed the old build log. Seems to be working now!

**Comment by Tobias-Fischer** [comment]
Note that this triggered a migration: https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/2392

On RoboStack, we have libopencv in the mutex package, so we should be safe ...

**Pull request merged**